### PR TITLE
fix: use different variables for vhd and e2e subscriptions

### PR DESCRIPTION
**Pull request by junjiezhang1997**
**What type of PR is this?**
/kind fix

**What this PR does / why we need it**:
use different variables for vhd and e2e subscriptions
**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [x] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:

```
none
```

### CODE REVIEW

File: e2e/windows/README.md [hunk 1]
```diff
@@ -78,7 +78,7 @@ windows 23H2 gen1 and windows 23H2 gen2.
 ## Generate and Update current windows test images
 
 Currently, we use the latest test images ""
-/subscriptions/$AZURE_BUILD_SUBSCRIPTION_ID/resourceGroups/$AZURE_BUILD_RESOURCE_GROUP_NAME/providers/Microsoft.Compute/galleries/$AZURE_BUILD_GALLERY_NAME/images/windows-e2e-test-$WINDOWS_E2E_IMAGE/versions/latest" (
+/subscriptions/$AZURE_E2E_IMAGE_SUBSCRIPTION_ID/resourceGroups/$AZURE_BUILD_RESOURCE_GROUP_NAME/providers/Microsoft.Compute/galleries/$AZURE_BUILD_GALLERY_NAME/images/windows-e2e-test-$WINDOWS_E2E_IMAGE/versions/latest" (
```
Comment (ShiqianTao) [inline e2e/windows/README.md:81]: The image definition name does not contain `e2e-test-`

File: e2e/windows/e2e-scenario.sh [hunk 2]
```diff
@@ -151,7 +151,7 @@ MC_WIN_VMSS_NAME=$(az vmss list -g $E2E_MC_RESOURCE_GROUP_NAME --query "[?contai
 VMSS_RESOURCE_Id=$(az resource show --resource-group $E2E_MC_RESOURCE_GROUP_NAME --name $MC_WIN_VMSS_NAME --resource-type Microsoft.Compute/virtualMachineScaleSets --query id --output tsv)
 
 az group export --resource-group $E2E_MC_RESOURCE_GROUP_NAME --resource-ids $VMSS_RESOURCE_Id --include-parameter-default-value > test.json
-IMAGE_REFERENCE="/subscriptions/$AZURE_BUILD_SUBSCRIPTION_ID/resourceGroups/$AZURE_BUILD_RESOURCE_GROUP_NAME/providers/Microsoft.Compute/galleries/$AZURE_BUILD_GALLERY_NAME/images/windows-$WINDOWS_E2E_IMAGE/versions/latest"
+IMAGE_REFERENCE="/subscriptions/$AZURE_E2E_IMAGE_SUBSCRIPTION_ID/resourceGroups/$AZURE_BUILD_RESOURCE_GROUP_NAME/providers/Microsoft.Compute/galleries/$AZURE_BUILD_GALLERY_NAME/images/windows-$WINDOWS_E2E_IMAGE/versions/latest"
```
Comment (ShiqianTao) [inline e2e/windows/e2e-scenario.sh:154]: Do you need to update resource group name or others?

File: e2e/windows/README.md (hunk 1 above)
Comment (junjiezhang1997) [reply e2e/windows/README.md]: Thanks. Updated.

File: e2e/windows/e2e-scenario.sh (hunk 2 above)
Comment (junjiezhang1997) [reply e2e/windows/e2e-scenario.sh]: Updated.

**Review by AbelHu — APPROVED** [review]
LGTM. Thanks for the PR.

**Pull request merged**